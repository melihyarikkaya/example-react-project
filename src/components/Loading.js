import React, { Component } from 'react';

class Loading extends Component {
  _setIsloading = () => {
    this.props.setIsloading(!this.props.isLoading);
  };

  render() {
    return (
      <div>
        <h3>
          isLoading State = <span className="state">{this.props.isLoading ? 'TRUE' : 'FALSE'}</span>
        </h3>
        <br />
        <button onClick={this._setIsloading}>
          Set isLoading to {!this.props.isLoading ? 'TRUE' : 'FALSE'}
        </button>
      </div>
    );
  }
}

export default Loading;
