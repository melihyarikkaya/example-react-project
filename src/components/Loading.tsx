import React, { Component } from 'react';

export interface LoadingStatic {
  setIsLoading(status: boolean): void;
  isLoading: boolean;
}

class Loading extends Component<LoadingStatic> {
  _setIsloading = () => {
    this.props.setIsLoading(!this.props.isLoading);
  };

  render() {
    return (
      <div>
        <h3>
          isLoading State = <span className="state">{this.props.isLoading ? 'TRUE' : 'FALSE'}</span>
        </h3>
        <br />
        <button onClick={this._setIsloading}>
          Set isLoading to {!this.props.isLoading ? 'TRUE' : 'FALSE'}
        </button>
      </div>
    );
  }
}

export default Loading;
