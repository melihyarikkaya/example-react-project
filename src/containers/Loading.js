import { connect } from 'react-redux';
import Loading from '../components/Loading';
import { setIsloading } from '../actions/actions';

const mapStateToProps = state => ({
  isLoading: state.loggerReducer.isLoading
});

const mapDispatchToProps = {
  setIsloading
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Loading);
