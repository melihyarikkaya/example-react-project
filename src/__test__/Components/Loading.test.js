import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';

import configureMockStore from 'redux-mock-store';
import loggerReducer from '../../reducers/logger-reducer'; // for get initial state

import Loading from '../../components/Loading';

describe('When Loading component rendered it', () => {
  let wrapper;
  let store;
  let mockStore;
  let fakeSetIsloading = jest.fn();
  let initialState = loggerReducer(undefined, {});

  beforeEach(() => {
    mockStore = configureMockStore();
    store = mockStore(initialState);
    wrapper = shallow(
      <Loading store={store} setIsloading={fakeSetIsloading} isLoading={initialState.isLoading} />
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match last snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Span text should be equal isLoading prop.', () => {
    it('If isLoading is false, span text should be FALSE', () => {
      expect(wrapper.find('h3>span').text()).toEqual('FALSE');
    });
    it('If isLoading is false, span text should be FALSE', () => {
      wrapper.setProps({ isLoading: true });
      expect(wrapper.find('h3>span').text()).toEqual('TRUE');
    });
  });

  it('should render the button', () => {
    expect(wrapper.exists('button')).toEqual(true);
  });

  it('should have an isLoading props', () => {
    const state = wrapper.instance().props.store.getState();
    expect(state.isLoading).toEqual(false);
  });

  it('should have a _setIsloading function', () => {
    expect(typeof wrapper.instance()._setIsloading).toEqual('function');
  });

  describe('When clicked button', () => {
    it('should trigger _setIsloading function', () => {
      let wrapperInstance = wrapper.instance();
      let stubbedFunc = sinon.stub(wrapperInstance, '_setIsloading').callsFake(() => {});
      wrapperInstance.forceUpdate();
      wrapper.find('button').simulate('click');
      expect(stubbedFunc.called).toEqual(true);
    });

    it('should trigger this.props.setIsloading redux action, when _setIsloading called', () => {
      let wrapperInstance = wrapper.instance();
      wrapperInstance._setIsloading();
      expect(wrapperInstance.props.setIsloading).toHaveBeenCalledWith(true);
    });
  });

  it('should render <span> tag which has text "FALSE"', () => {
    expect(wrapper.find('.state').text()).toEqual('FALSE');
  });
});
