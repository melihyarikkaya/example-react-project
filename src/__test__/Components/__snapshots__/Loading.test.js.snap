// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Loading component rendered it should match last snapshot 1`] = `
<div>
  <h3>
    isLoading State = 
    <span
      className="state"
    >
      FALSE
    </span>
  </h3>
  <br />
  <button
    onClick={[Function]}
  >
    Set isLoading to 
    TRUE
  </button>
</div>
`;
